Add/user licensing banner

oikeme opened the pull request:
#### Changes proposed in this Pull Request

Add a banner to the checkout flow and pricing page that informs a user when they have unused licenses that can be activated.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Ensure that the banner adjusts and is displayed correctly for the mobile experience with both the Jetpack Connect flow and the plans page flow. 

- Download the PR
- Start both Calypso and Jetpack Cloud ```yarn start``` and ```yarn start-jetpack-cloud-p``` in order
- Ensure that you have a purchase for a plan that has a license that has not yet been activated or associated with a site. 

Plans Page Flow
- Create a new JN site and connect the site. 
- Once the site is connected go to the plans page. Once on the plans page update the url from https://cloud.jetpack.com to http://jetpack.cloud.localhost:3001 
-  Ensure that the banner is present and adjust correctly for mobile. 

Jetpack Connect Flow
- Attempt to connect a disconnected site. Once on the page to authorize the connection update the url from wordpress.com to http://calypso.localhost:3000/
- Authorize the connection and ensure that you are taken to the pricing page and that the banner indicating an available license can be activated is present. Ensure that the banner adjust correctly for mobile


<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

rcanepa committed:
Create new dialog component to prompt the user

rcanepa committed:
Disable post-connection redirection

rcanepa committed:
Add the modal content

rcanepa committed:
Pass down the redirect target

rcanepa committed:
Refactor logic that extract the license

rcanepa committed:
Move query components up to the parent component

rcanepa committed:
Compute the site admin URL in the component

rcanepa committed:
Show the modal under the right conditions

rcanepa committed:
Improve the appearance of paragraphs in cloud

rcanepa committed:
Make it mobile friendly

rcanepa committed:
Add an X close button

rcanepa committed:
Track view and click events

rcanepa committed:
Wrap fn to close in useCallback

rcanepa committed:
Update test to make it match current behavior

rcanepa committed:
Update test snapshot

oikeme committed:
Add banner for license activation

Add a banner in the checkout process to indicate when there are licenses that are available and have not yet been activated

oikeme committed:
Add lock icon

Add the key icon with the correct color and update the icon name

rcanepa commented on client/my-sites/plans/jetpack-plans/selector.tsx:
```diff
@@ -188,6 +198,8 @@ const SelectorPage: React.FC< SelectorPageProps > = ( {
 					options={ { useJetpackGoogleAnalytics: ! isJetpackCloud() } }
 				/>
 
+				<LicensingActivationBanner siteId={ siteId } />
```
You're missing checking the `siteId && enableUserLicensesDialog`. Without that, the banner is always rendered.

rcanepa commented on client/components/jetpack/licensing-activation-banner/index.tsx:
```diff
@@ -0,0 +1,32 @@
+			<div className="licensing-activation-banner">
+				<div className="licensing-activation-banner__copy">
+					<img className="licensing-activation-banner__key-icon" src={ keyIcon } alt="" />
+					{ translate( 'You have an available product license key ' ) }
```
We need to add a `.` at the end of the sentence.

rcanepa commented on client/components/jetpack/licensing-activation-banner/index.tsx:
```diff
@@ -0,0 +1,32 @@
+	return (
+		<>
+			<div className="licensing-activation-banner">
+				<div className="licensing-activation-banner__copy">
```
I don't think we need two levels of `div` tags.

rcanepa commented:
I made the PR point to my `add/user-licensing-modal` branch instead of `trunk`. That way, it is easier to understand your changes in isolation.

----

**Comments**

**The key icon doesn't look aligned to the copy and we are missing a `.` at the end of the sentence, as shown in the following capture.**

![image](https://user-images.githubusercontent.com/3418513/148770347-d4a840a1-f449-4edf-8959-64eaa575237d.png)

I think that the alignment issue can be fixed by relying on `display: flex;`.

**On Jetpack Connect, the banner is too close to the Jetpack logo.**

![image](https://user-images.githubusercontent.com/3418513/148770506-caff52f4-843b-478b-9b90-d0edec4beea5.png)

**The banner appears regardless of having available licenses**

We need to include the same checks that control the dialog/modal appareance.

oikeme committed:
Update the activate now banner

Update the key icon viewbox to align it with the banner text. Also remove unnecessary nested div and fix the copy on the banner.

oikeme committed:
Update banner css

Remove unused css and update banner padding

oikeme committed:
Update banner css and display conditions

Update the activate a license banner css for mobile and Jetpack connect and fix conditions under which the banner displays.

vitozev commented on client/components/jetpack/licensing-activation-banner/index.tsx:
```diff
@@ -0,0 +1,37 @@
+			<>
+				<div className="licensing-activation-banner">
+					<img className="licensing-activation-banner__key-icon" src={ keyIcon } alt="" />
+					{ translate( 'You have an available product license key. ' ) }
```
Is it intentional to have that extra space at the end of this string? It looks like the "Activate it now" text appearance isn't depending on it.

oikeme replied:
This was not intentional and has been updated.

vitozev commented on client/components/jetpack/licensing-activation-banner/style.scss:
```diff
@@ -0,0 +1,45 @@
+.licensing-activation-banner {
+	padding-top: 16px;
+    padding-bottom: 16px;
+    color: var( --studio-white );
```
The text color in design is `#DCDCDE`, although it looks like we don't have such a variable available. @ederrengifo, should we update the color to match the design?

rcanepa replied:
We need to be careful with any decision because the banner is displayed in two different enviroments: cloud.jetpack.com/pricing and wordpress.com/jetpack/connect/plans.

ederrengifo replied:
I will recommend using the closer variable to that value. In our Figma library that color is called `grey 5` by the way. If that's a problem because of the two different environments, then I would recommend using white by default.

vitozev commented on client/components/jetpack/licensing-activation-banner/style.scss:
```diff
@@ -0,0 +1,45 @@
+	background-color: var( --studio-black );
+
+	@include breakpoint-deprecated( '>660px' ) {
+        display: flex;
+        justify-content: center;
+        align-items: center;
```
I think we can apply these properties even for devices `<660px` (i.e. move it outside the media query), with one small addition: `flex-wrap: wrap` (it will cause the flex items to wrap once they become too wide for the container). 

Before | After
--- | ---
![Screen Capture on 2022-01-11 at 10-50-57](https://user-images.githubusercontent.com/1749918/148913596-44945e69-acfb-48ba-b8c6-bb9eb9ca2c86.gif) | ![Screen Capture on 2022-01-11 at 10-53-36](https://user-images.githubusercontent.com/1749918/148913662-461179ab-aceb-44b1-a005-d3efdd205262.gif)

rcanepa replied:
This is a nice suggestion.

vitozev reviewed (commented):
All the functional tests are working as expected! I just left a few minor style-related comments.

oikeme committed:
Update license banner text

Remove the extra space at the end of the license banner text

oikeme committed:
Update style.scss

Update css for license activation banner for mobile and to properly wrap banner text.

oikeme committed:
Merge branch 'trunk' into add/user-licensing-banner

oikeme committed:
Update style.scss

Fix spacing from merge

rcanepa reviewed (approved):
LGTM.

Awesome job, Oli!

oikeme committed:
Update style.scss

Update text color

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7079494

Hi @oikeme, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Activate it now`
- `You have an available product license key.`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7079494).
